soh/src/code/z_player_lib.c

```diff
@@ -1068,13 +1094,30 @@ s32 func_800902F0(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s
             *dList = sArmOutDLs[gSaveContext.linkAge];
         } else if (limbIndex == PLAYER_LIMB_L_HAND) {
             *dList = sHandOutDLs[gSaveContext.linkAge];
+
+            if(CVarGetInteger("gBowSlingShotAmmoFix", 0)){
+                if (Player_HoldsSlingshot(this)) {
+                    *dList = NULL;
+                } else if (LINK_IS_ADULT && (Player_HoldsBow(this) || Player_HoldsSlingshot(this))) {
+                    *dList = gLinkAdultRightHandOutNearDL;
+                }
+            }
         } else if (limbIndex == PLAYER_LIMB_R_SHOULDER) {
             *dList = sRightShoulderNearDLs[gSaveContext.linkAge];
         } else if (limbIndex == PLAYER_LIMB_R_FOREARM) {
             *dList = D_80125F30[gSaveContext.linkAge];
         } else if (limbIndex == PLAYER_LIMB_R_HAND) {
-            *dList = Player_HoldsHookshot(this) ? gLinkAdultRightHandHoldingHookshotFarDL
-                                                : sHoldingFirstPersonWeaponDLs[gSaveContext.linkAge];
+            *dList = sHoldingFirstPersonWeaponDLs[gSaveContext.linkAge];
+
+            if (Player_HoldsHookshot(this)) {
+                *dList = gLinkAdultRightHandHoldingHookshotFarDL;
+            } else if (CVarGetInteger("gBowSlingShotAmmoFix", 0)) {
+                if(Player_HoldsBow(this)) {
+                    *dList = gLinkAdultRightHandHoldingBowFirstPersonDL;
+                } else if (Player_HoldsSlingshot(this)) {
+                    *dList = gLinkChildRightArmStretchedSlingshotDL;
+                }
+            }
```

Comment:
functionally this will be the same, but i don't like the change to authentic behavior from setting once to setting twice, so instead this could be
```suggestion
            s32 firstPersonWeaponIndex = gSaveContext.linkAge;
            if (CVarGetInteger("gBowSlingShotAmmoFix", 0)) {
                if (Player_HoldsBow(this)) {
                    firstPersonWeaponIndex = 0;
                } else if (Player_HoldsSlingshot(this)) {
                    firstPersonWeaponIndex = 1;
                }
            }
            *dList = Player_HoldsHookshot(this) ? gLinkAdultRightHandHoldingHookshotFarDL
                                                                       : sHoldingFirstPersonWeaponDLs[firstPersonWeaponIndex];
```